docs/guide/cache/content_aware_cache.md

```diff
@@ -293,3 +350,228 @@ bin/console fos:httpcache:invalidate:tag ez-all
 
     Similarly to purging from code, the tags you purge on, are prefixed to match the currently configured SiteAccess. 
     When you use this command in combination with multi-repository setup, make sure to specify SiteAccess argument.
+
+## Testing and Debugging HTTP cache
+
+It is important to test your code in an environment which is as similar as your production environment as possible. That
+means that if only are testing locally using the default Symfony Reverse proxy when your are going to use Varnish or
+Fastly in production, you are likely edning up some (bad) surprises. Due to the symfony reverse proxy's lack of support for ESIs, it behaves
+quite different from Varnish and Fastly in some aspects.
+If you are going to use Varnish in production, make sure you also test your code with Varnish.
+If you are going to use Fastly in production, testing with Fastly in your developer install is likely not feasible
+(you're local development environment must then be accessible for Fastly). Testing with Varnish instead will in most
+cases do the job. But if you need to change the varnish configuration to make your site work, be aware that Varnish and Fastly uses different dialects, and
+that .vcl code for Varnish V6.x will likely not work as-is on Fastly.
+
+In this chapter we'll look into how to debug problems related to HTTP-cache. In order to that we must be able to look at
+the responses and headers [[= product_name =]] sends to the HTTP Cache, and not so much on what responses and headers
+the HTTP Cache sends the client (web-browser).
+That means that you must be able to send requests to your origin (webserver), requests that does *not* go through Varnish/Fastly.
+If you run nginx and Varnish on premise, you likely know what host and port number both Varnish and nginx runs on. If you
+are testing on a Fastly enabled environment on Ibexa Cloud provided by Platform.sh, you'll need to use the Platform.sh
+Dashboard in order to obtain the endpoint for nginx.
+
+In the example below we'll debug and figure out why a given frontpage is not cached properly by Fastly. We have learned
+that by running the following command *multiple* times:
+
+```curl -IXGET https://www.staging.foobar.com.us-2.platformsh.site```
+
+And it always outputs
+
+```
+HTTP/2 200
+(...)
+x-cache: MISS
+```
+
+### Finding the nginx endpoint on Platform.sh
+
+Assuming you on Platform have a project with id=`asd123asd123` and an environment named `Staging` which you access via Fastly on
```

Comment:
@Oc7at : Do you have some input for a straight forward process for how to get the route to nginx endpoing on a fastly-enabled environment? I found it to be slightly different depending on if you are on grid vs dedicated etc ( on dedicated, i think I in the past found the region host (......eu2.platformsh.site or something), and then just added `gw.` in front...

Reply:
If you're looking for the origin (or "backend" as it's the term used by Fastly), you'll indeed want `c.<clusterid>.ent.platform.sh` for dedicated clusters. For non-dedicated projects, you can use [our documentation page](https://docs.platform.sh/development/public-ips.html) that lists all the IPs of our regions; you can usually use `gw.<region>.platformsh.site` though, like `gw.eu-2.platformsh.site` as you mention. Another good way of doing that for non-dedicated environments with our CLI is to use `platform environment:info edge_hostname`, as stated [on this page](https://docs.platform.sh/domains/steps/dns.html?#where-should-the-cname-point-to).